src/OpenTelemetry.OpAmp.Client/Listeners/Messages/AgentIdentificationMessage.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpAmp.Protocol;
+
+namespace OpenTelemetry.OpAmp.Client.Listeners.Messages;
+
+internal class AgentIdentificationMessage : IOpAmpMessage
+{
+    public AgentIdentificationMessage(AgentIdentification agentIdentification)
+    {
+        this.AgentIdentification = agentIdentification;
+    }
+
+    public AgentIdentification AgentIdentification { get; set; }
+}
```

Comment:
Could make all these be a bit less boiler-plate-y with primary constructors?

```suggestion
internal class AgentIdentificationMessage(AgentIdentification agentIdentification) : IOpAmpMessage
{
    public AgentIdentification AgentIdentification { get; set; } = agentIdentification;
}
```